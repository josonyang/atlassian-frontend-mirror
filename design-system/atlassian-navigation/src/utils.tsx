/**
 * Returns a new object with only non-empty properties of the input object.
 *
 * In this context, "empty" refers to properties with `null`, `undefined`, or `""`
 * (empty string) values.
 *
 * @param obj - The input object.
 * @returns A new object with non-empty properties of the input object.
 */
export function stripEmptyProperties<T extends object>(obj: T): Partial<T> {
	return Object.entries(obj).reduce((acc: Partial<T>, [key, value]) => {
		if (value !== null && value !== undefined && value !== '') {
			acc[key as keyof T] = value;
		}
		return acc;
	}, {} as Partial<T>);
}
