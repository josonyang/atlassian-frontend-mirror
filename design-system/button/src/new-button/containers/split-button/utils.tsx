import { Children, type ReactNode } from 'react';

export const getActions = (children: ReactNode) => {
	const [PrimaryAction, SecondaryAction] = Children.toArray(children);

	if (
		typeof process !== 'undefined' &&
		process.env.NODE_ENV === 'development' &&
		(!PrimaryAction || !SecondaryAction)
	) {
		// TODO: i18n?
		throw new SyntaxError('SplitButton requires two children to be provided');
	}

	return { PrimaryAction, SecondaryAction };
};
