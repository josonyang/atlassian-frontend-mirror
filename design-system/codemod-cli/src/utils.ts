/* Utility functions to be used in codemod-cli. */

const returnLineEnding = (source: string) => {
  var cr = source.split('\r').length;
  var lf = source.split('\n').length;
  var crlf = source.split('\r\n').length;
  if (cr + lf === 0) {
    return 'NONE';
  }
  if (crlf === cr && crlf === lf) {
    return 'CRLF';
  }
  if (cr > lf) {
    return 'CR';
  } else {
    return 'LF';
  }
};

const getLineEndingRegex = (type: string) => {
  if (['CR', 'LF', 'CRLF'].indexOf(type) === -1) {
    throw new Error(
      "Line ending '" + type + "' is not supported, use CR, LF or CRLF",
    );
  }
  if (type === 'LF') {
    return '\n';
  }
  if (type === 'CR') {
    return '\r';
  }
  if (type === 'CRLF') {
    return '\r\n';
  }
};

export const fixLineEnding = (source: string, lineEnding: string) => {
  const current = returnLineEnding(source);
  if (current === lineEnding) {
    return source;
  }
  const regexCurrentLineEnding = getLineEndingRegex(current);
  const regexLineEnding = getLineEndingRegex(lineEnding);
  if (current && regexLineEnding && regexCurrentLineEnding) {
    return source.replace(
      new RegExp(regexCurrentLineEnding, 'g'),
      regexLineEnding,
    );
  }
  return source;
};
