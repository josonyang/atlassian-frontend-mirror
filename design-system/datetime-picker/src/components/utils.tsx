// https://github.com/date-fns/date-fns-upgrade/blob/master/src/v2/convertTokens/index.ts
type TokensMap = {
  [v1token: string]: string;
};

const tokensMap: TokensMap = {
  // 'D MMMM': '',
  // 'Do MMMM': '',
  // 'DD MMMM': '',
  M: 'L',
  Mo: 'Mo',
  MM: 'LL',
  MMM: 'LLL',
  MMMM: 'LLLL',
  Q: 'q',
  Qo: 'qo',
  D: 'd',
  Do: 'do',
  DD: 'dd',
  DDD: 'D',
  DDDo: 'Do',
  DDDD: 'DDD',
  d: 'i',
  do: 'io',
  dd: 'iiiiii',
  ddd: 'iii',
  dddd: 'iiii',
  A: 'a',
  a: 'a',
  aa: 'aaaa',
  E: 'i',
  W: 'I',
  Wo: 'Io',
  WW: 'II',
  YY: 'yy',
  YYYY: 'yyyy',
  GG: 'RR',
  GGGG: 'RRRR',
  H: 'H',
  HH: 'HH',
  h: 'h',
  hh: 'hh',
  m: 'm',
  mm: 'mm',
  s: 's',
  ss: 'ss',
  S: 'S',
  SS: 'SS',
  SSS: 'SSS',
  Z: 'xxx',
  ZZ: 'xx',
  X: 't',
  x: 'T',
} as const;

const v1tokens = Object.keys(tokensMap).sort().reverse();

const tokensRegExp = new RegExp(
  // v1 escape string (unsure the purpose of post-pipe capture group)
  '(\\[[^\\[]*\\])|(\\\\)?' +
    // v2 escape string
    "((?<=').+(?=')|" +
    // All v1 tokens
    v1tokens.join('|') +
    '|.)',
  'g',
);

type TokensBuffer = {
  formatBuffer: string[];
  escapedTextBuffer: string[];
};

export function convertTokens(format: string): string {
  const tokensCaptures = format.match(tokensRegExp);
  if (!tokensCaptures) {
    return format;
  }

  return tokensCaptures
    .reduce(
      (parsed, tokenString, index) => {
        const v2token = tokensMap[tokenString];

        if (!v2token) {
          const escapedCaptures = tokenString.match(/^\[(.+)\]$/);
          if (escapedCaptures) {
            parsed.escapedTextBuffer.push(escapedCaptures[1]);
          } else {
            parsed.escapedTextBuffer.push(tokenString);
          }
        }

        const endOfString = index === tokensCaptures.length - 1;
        if (parsed.escapedTextBuffer.length && (v2token || endOfString)) {
          // This allows double parentheses to be rendered correctly
          // according to date-fns's spec.
          // https://date-fns.org/v2.29.3/docs/format
          const filteredEscapedTextBuffer = parsed.escapedTextBuffer
            .filter((token) => token !== "'")
            .join('');
          parsed.formatBuffer.push(`'${filteredEscapedTextBuffer}'`);
          parsed.escapedTextBuffer = [];
        }

        if (v2token) {
          parsed.formatBuffer.push(v2token);
        }

        return parsed;
      },
      { formatBuffer: [], escapedTextBuffer: [] } as TokensBuffer,
    )
    .formatBuffer.join('');
}
