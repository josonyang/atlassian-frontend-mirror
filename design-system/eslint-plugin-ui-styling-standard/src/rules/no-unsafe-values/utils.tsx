import type { Rule } from 'eslint';
import type { Variable } from 'eslint-scope';
import type {
	CallExpression,
	Expression,
	Identifier,
	Node,
	ObjectExpression,
	Property,
} from 'estree-jsx';

import {
	type AllowList,
	getAllowedDynamicKeys,
	getAllowedFunctionCalls,
	isAllowListedVariable,
} from '@atlaskit/eslint-utils/allowed-function-calls';
import { findIdentifierNode } from '@atlaskit/eslint-utils/find-identifier-node';
import { findVariable } from '@atlaskit/eslint-utils/find-variable';

import type { MessageId } from './messages';

function isReferenceToCssVar(variable: Variable) {
	const definitions = variable.defs;

	return definitions.every((definition) => {
		if (definition.type !== 'Variable') {
			return false;
		}

		const initializer = definition.node.init;
		if (!initializer) {
			return false;
		}

		return (
			initializer.type === 'Literal' &&
			typeof initializer.value === 'string' &&
			initializer.value.startsWith('--')
		);
	});
}

export class Linter {
	private readonly allowedDynamicKeys: AllowList;
	private readonly allowedFunctionCalls: AllowList;

	constructor(
		private readonly context: Rule.RuleContext,
		private readonly baseNode: CallExpression,
	) {
		this.allowedDynamicKeys = getAllowedDynamicKeys(context.options);
		this.allowedFunctionCalls = getAllowedFunctionCalls(context.options);
	}

	/**
	 * Wrapper around `context.report` to provide type checking on the message ID.
	 */
	private report(descriptor: { node: Node; messageId: MessageId }) {
		this.context.report(descriptor);
	}

	/**
	 * Fallback case to lint any identifier that is not otherwise handled.
	 */
	private lintUnhandledIdentifier(identifier: Identifier) {
		this.report({ node: identifier, messageId: 'no-variables' });
	}

	private lintIdentifier(value: Identifier) {
		if (value.name === 'undefined') {
			/**
			 * `undefined` gets caught implicitly anyway by other checks below,
			 * but we are explicitly handling it here.
			 *
			 * We could potentially introduce a distinct message for it as well.
			 */
			this.lintUnhandledIdentifier(value);
			return;
		}

		const variable = findVariable({
			identifier: value,
			sourceCode: this.context.getSourceCode(),
		});
		if (!variable) {
			this.lintUnhandledIdentifier(value);
			return;
		}

		/**
		 * Allows using whitelisted imported values
		 */
		if (
			isAllowListedVariable({
				allowList: this.allowedFunctionCalls,
				variable,
			})
		) {
			return;
		}

		const definitions = variable.defs;
		if (definitions.length === 0) {
			this.lintUnhandledIdentifier(value);
			return;
		}

		definitions.forEach((definition) => {
			if (definition.type === 'Variable') {
				const initializer = definition.node.init;
				if (!initializer) {
					this.lintUnhandledIdentifier(value);
					return;
				}

				/**
				 * Variables can be used if they resolve to a literal.
				 */
				if (initializer.type === 'Literal') {
					return;
				}

				/**
				 * Not allowing tagged template expressions through variables.
				 */
				if (initializer.type === 'TaggedTemplateExpression') {
					this.report({ node: value, messageId: 'no-function-calls' });
					return;
				}

				/**
				 * Variables can be used if they resolve to an allowed function call.
				 */
				if (initializer.type === 'CallExpression') {
					if (this.isAllowedFunctionCall(initializer)) {
						return;
					}
					this.report({ node: value, messageId: 'no-function-calls' });
					return;
				}

				if (initializer.type === 'TemplateLiteral') {
					for (const expression of initializer.expressions) {
						this.lintValue(expression);
					}
					return;
				}

				/**
				 * This branch is only here for clarity.
				 *
				 * We could fallback to `no-variables` for it too.
				 */
				if (initializer.type === 'ObjectExpression') {
					this.report({ node: value, messageId: 'no-object-references' });
					return;
				}
			}

			this.lintUnhandledIdentifier(value);
		});
	}

	private isAllowedDynamicKey(expression: Expression) {
		const identifier = findIdentifierNode(expression);
		if (!identifier) {
			return false;
		}

		const variable = findVariable({
			identifier,
			sourceCode: this.context.getSourceCode(),
		});
		if (!variable) {
			return false;
		}

		return (
			isReferenceToCssVar(variable) ||
			isAllowListedVariable({ allowList: this.allowedDynamicKeys, variable })
		);
	}

	private lintKey(property: Property) {
		/**
		 * If it's not computed then it must be a plain string.
		 *
		 * Private identifiers cannot be computed,
		 * but we need the extra check for type narrowing.
		 */
		if (!property.computed || property.key.type === 'PrivateIdentifier') {
			return;
		}

		/**
		 * Some dynamic keys are explicitly allowed.
		 */
		if (this.isAllowedDynamicKey(property.key)) {
			return;
		}

		this.report({ node: property.key, messageId: 'no-dynamic-keys' });
	}

	private isAllowedFunctionCall(value: CallExpression) {
		const identifier = findIdentifierNode(value);
		if (!identifier) {
			return false;
		}

		const variable = findVariable({
			identifier,
			sourceCode: this.context.getSourceCode(),
		});
		if (!variable) {
			return false;
		}

		return isAllowListedVariable({
			allowList: this.allowedFunctionCalls,
			variable,
		});
	}

	private lintFunctionCall(value: CallExpression) {
		if (this.isAllowedFunctionCall(value)) {
			return;
		}

		this.report({ node: value, messageId: 'no-function-calls' });
	}

	private lintObjectExpression(expression: ObjectExpression) {
		for (const property of expression.properties) {
			if (property.type === 'SpreadElement') {
				this.report({
					node: property,
					messageId: 'no-spread-elements',
				});
			} else {
				this.lintKey(property);
				this.lintValue(property.value);
			}
		}
	}

	private lintUnhandledValue(value: Node) {
		/**
		 * Default to blocking value if it isn't explictly handled.
		 */
		this.report({ node: value, messageId: 'no-unsafe-values' });
	}

	private lintValue(value: Property['value']): void {
		/**
		 * Literals are always allowed by this rule.
		 */
		if (value.type === 'Literal') {
			return;
		}

		/**
		 * Allow negative numbers.
		 */
		if (
			value.type === 'UnaryExpression' &&
			value.operator === '-' &&
			value.argument.type === 'Literal'
		) {
			return;
		}

		if (value.type === 'TemplateLiteral') {
			for (const expression of value.expressions) {
				this.lintValue(expression);
			}
			return;
		}

		if (value.type === 'ConditionalExpression' || value.type === 'LogicalExpression') {
			this.report({ node: value, messageId: 'no-conditional-values' });
			return;
		}

		if (value.type === 'BinaryExpression') {
			this.lintValue(value.left);
			this.lintValue(value.right);
			return;
		}

		if (value.type === 'ObjectExpression') {
			this.lintObjectExpression(value);
			return;
		}

		if (value.type === 'Identifier') {
			this.lintIdentifier(value);
			return;
		}

		if (value.type === 'TaggedTemplateExpression') {
			this.report({ node: value, messageId: 'no-function-calls' });
			return;
		}

		if (value.type === 'CallExpression') {
			this.lintFunctionCall(value);
			return;
		}

		if (value.type === 'ArrowFunctionExpression' || value.type === 'FunctionExpression') {
			/**
			 * Function values are handled by `no-dynamic-styles` instead
			 */
			return;
		}

		if (value.type === 'MemberExpression') {
			this.report({ node: value, messageId: 'no-object-access' });
			return;
		}

		if (value.type === 'ArrayExpression') {
			this.report({ node: value, messageId: 'no-array-values' });
			return;
		}

		this.lintUnhandledValue(value);
	}

	private lintArgument(argument: CallExpression['arguments'][number]) {
		if (argument.type === 'ArrowFunctionExpression') {
			if (argument.body.type === 'ObjectExpression') {
				// Something in this form:
				//
				// const styles = styled.div(
				//     (props) => ({ color: props.height }),
				// );
				this.lintArgument(argument.body);
			}
			return;
		}

		if (argument.type === 'ArrayExpression') {
			for (const element of argument.elements) {
				if (element === null) {
					continue;
				}
				this.lintArgument(element);
			}
			return;
		}

		if (argument.type === 'ObjectExpression') {
			this.lintObjectExpression(argument);
			return;
		}

		if (argument.type === 'CallExpression') {
			this.lintFunctionCall(argument);
			return;
		}

		if (argument.type === 'ConditionalExpression' || argument.type === 'LogicalExpression') {
			this.report({ node: argument, messageId: 'no-conditional-values' });
			return;
		}

		if (argument.type === 'Identifier' || argument.type === 'MemberExpression') {
			this.report({ node: argument, messageId: 'no-identifier-arguments' });
			return;
		}

		this.lintUnhandledValue(argument);
	}

	private lintArguments(args: CallExpression['arguments']) {
		for (const argument of args) {
			this.lintArgument(argument);
		}
	}

	run(): void {
		this.lintArguments(this.baseNode.arguments);
	}
}
