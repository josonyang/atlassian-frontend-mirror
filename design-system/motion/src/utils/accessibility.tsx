import { useEffect, useState } from 'react';

import { bind } from 'bind-event-listener';

const isMatchMediaAvailable = (): boolean =>
	typeof window !== 'undefined' && 'matchMedia' in window;

/**
 * Use for any programatic motions needed at runtime.
 * Will return `true` if the current user prefers reduced motion.
 * This is generally set through OS preferences/settings.
 */
export const isReducedMotion = (): boolean => {
	if (!isMatchMediaAvailable()) {
		return false;
	}

	const { matches } = window.matchMedia('(prefers-reduced-motion: reduce)');
	return matches;
};

/**
 * A React hook version of {@link isReducedMotion}.
 * Useful for React components that need to re-render if the user's motion
 * preference changes at runtime.
 */
export const useIsReducedMotion = (): boolean => {
	const [prefersReducedMotion, setPrefersReducedMotion] = useState(isReducedMotion);

	useEffect(() => {
		if (!isMatchMediaAvailable()) {
			return;
		}

		const mediaQueryList = window.matchMedia('(prefers-reduced-motion: reduce)');

		const onChange = (event: MediaQueryListEvent) => setPrefersReducedMotion(event.matches);

		return bind(mediaQueryList, {
			type: 'change',
			listener: onChange,
		});
	}, []);

	return prefersReducedMotion;
};

// Ticket for adding an ESLint rule to recommend that any `animation` or `transition` styles are
// disabled based on the user's motion preference: https://product-fabric.atlassian.net/browse/DSP-23842
// eslint-disable-next-line @repo/internal/deprecations/deprecation-ticket-required
/**
 * Use for any CSS based motion (animation or transition).
 * Always put at the end of your declaration for correct use of the cascade.
 * Reduced motion preference is generally set through OS preferences/settings.
 *
 * @deprecated This is not fully compatible with Compiled CSS and will be removed in the future.
 * You should hardcode the `prefers-reduced-motion` media query in your file instead.
 */
export const reduceMotionAsPerUserPreference = {
	'@media (prefers-reduced-motion: reduce)': {
		animation: 'none',
		transition: 'none',
	},
} as const;

/**
 * @deprecated {@link https://hello.atlassian.net/browse/ENGHEALTH-4709 Internal documentation for deprecation (no external access)}
 * This is not compatible with Compiled CSS and will be removed in the future.
 * You should hardcode the `prefers-reduced-motion` media query in your file instead.
 */
export const prefersReducedMotion = () => reduceMotionAsPerUserPreference;
