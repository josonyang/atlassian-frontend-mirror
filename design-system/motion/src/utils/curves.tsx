export type AnimationCurve =
	| 'cubic-bezier(0.15,1,0.3,1)'
	| 'cubic-bezier(0.2,0,0,1)'
	| 'cubic-bezier(0.8,0,0,0.8)'
	| 'cubic-bezier(0.4,0,0,1)'
	| 'cubic-bezier(0.6,0,0,1)'
	| 'cubic-bezier(0.8,0,0,1)'
	| 'cubic-bezier(0,0,1,1)';

export const easeInOut: AnimationCurve = 'cubic-bezier(0.15,1,0.3,1)';

export const easeOut: AnimationCurve = 'cubic-bezier(0.2,0,0,1)';

export const easeIn: AnimationCurve = 'cubic-bezier(0.8,0,0,0.8)';

export const easeIn40Out: AnimationCurve = 'cubic-bezier(0.4,0,0,1)';

export const easeIn60Out: AnimationCurve = 'cubic-bezier(0.6,0,0,1)';

export const easeIn80Out: AnimationCurve = 'cubic-bezier(0.8,0,0,1)';

export const linear: AnimationCurve = 'cubic-bezier(0,0,1,1)';
