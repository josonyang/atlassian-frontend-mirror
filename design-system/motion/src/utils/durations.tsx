/**
 * Think of this as the motion equivalent of the @atlaskit/theme `grid()`.
 */
export const durationStep = 25;

export type Durations = 'none' | 'small' | 'medium' | 'large';

export const durations: Record<Durations, number> = {
	none: 0,
	small: 100,
	medium: 350,
	large: 700,
};

/**
 * Used to multiply the initial duration for exiting motions.
 */
const EXITING_MOTION_MULTIPLIER = 0.5;

export const exitingDurations: Record<Durations, number> = {
	none: durations.none,
	small: durations.small * EXITING_MOTION_MULTIPLIER,
	medium: durations.medium * EXITING_MOTION_MULTIPLIER,
	large: durations.large * EXITING_MOTION_MULTIPLIER,
};
