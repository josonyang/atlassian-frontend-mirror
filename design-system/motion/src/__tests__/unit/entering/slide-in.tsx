import React from 'react';

import { render, screen } from '@testing-library/react';

import ExitingPersistence from '../../../entering/exiting-persistence';
import SlideIn from '../../../entering/slide-in';
import { easeIn, easeOut } from '../../../utils/curves';

jest.mock('../../../utils/accessibility');

describe('<SlideIn />', () => {
	it('should default to medium duration', () => {
		render(
			<SlideIn enterFrom="left">{(props) => <div data-testid="target" {...props} />}</SlideIn>,
		);

		expect(screen.getByTestId('target')).toHaveCompiledCss('animation-duration', '.35s');
	});

	it('should override default duration', () => {
		render(
			<SlideIn duration="small" enterFrom="left">
				{(props) => <div data-testid="target" {...props} />}
			</SlideIn>,
		);

		expect(screen.getByTestId('target')).toHaveCompiledCss('animation-duration', '.1s');
	});

	it('should slide in easing out', () => {
		render(
			<SlideIn enterFrom="left">{(props) => <div data-testid="target" {...props} />}</SlideIn>,
		);

		expect(screen.getByTestId('target')).toHaveCompiledCss(
			'animation-timing-function',
			easeOut.replace('0.', '.'),
		);
	});

	it('should slide out easing in', () => {
		const { rerender } = render(
			<ExitingPersistence>
				<SlideIn enterFrom="left">{(props) => <div data-testid="target" {...props} />}</SlideIn>
			</ExitingPersistence>,
		);

		rerender(<ExitingPersistence>{false}</ExitingPersistence>);

		expect(screen.getByTestId('target')).toHaveCompiledCss(
			'animation-timing-function',
			easeIn.replace(new RegExp(/0\./, 'g'), '.'),
		);
	});
});
