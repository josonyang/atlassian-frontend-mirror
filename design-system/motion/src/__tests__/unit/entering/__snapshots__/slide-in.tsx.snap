// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlideIn /> should animate entering from bottom with fade in 1`] = `
{
  "0%": {
    "opacity": 0,
    "transform": "translate3d(0, 100%, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from bottom with fade inout 1`] = `
{
  "0%": {
    "opacity": 0,
    "transform": "translate3d(0, 100%, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from bottom with fade none 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "translate3d(0, 100%, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from bottom with fade out 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "translate3d(0, 100%, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from left with fade in 1`] = `
{
  "0%": {
    "opacity": 0,
    "transform": "translate3d(-100%, 0, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from left with fade inout 1`] = `
{
  "0%": {
    "opacity": 0,
    "transform": "translate3d(-100%, 0, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from left with fade none 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "translate3d(-100%, 0, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from left with fade out 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "translate3d(-100%, 0, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from right with fade in 1`] = `
{
  "0%": {
    "opacity": 0,
    "transform": "translate3d(100%, 0, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from right with fade inout 1`] = `
{
  "0%": {
    "opacity": 0,
    "transform": "translate3d(100%, 0, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from right with fade none 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "translate3d(100%, 0, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from right with fade out 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "translate3d(100%, 0, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from top with fade in 1`] = `
{
  "0%": {
    "opacity": 0,
    "transform": "translate3d(0, -100%, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from top with fade inout 1`] = `
{
  "0%": {
    "opacity": 0,
    "transform": "translate3d(0, -100%, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from top with fade none 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "translate3d(0, -100%, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate entering from top with fade out 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "translate3d(0, -100%, 0)",
  },
  "100%": {
    "opacity": 1,
    "transform": "none",
  },
}
`;

exports[`<SlideIn /> should animate exiting from bottom with fade in 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 1,
    "transform": "translate3d(0, 100%, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from bottom with fade inout 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 0,
    "transform": "translate3d(0, 100%, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from bottom with fade none 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 1,
    "transform": "translate3d(0, 100%, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from bottom with fade out 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 0,
    "transform": "translate3d(0, 100%, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from left with fade in 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 1,
    "transform": "translate3d(-100%, 0, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from left with fade inout 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 0,
    "transform": "translate3d(-100%, 0, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from left with fade none 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 1,
    "transform": "translate3d(-100%, 0, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from left with fade out 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 0,
    "transform": "translate3d(-100%, 0, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from right with fade in 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 1,
    "transform": "translate3d(100%, 0, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from right with fade inout 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 0,
    "transform": "translate3d(100%, 0, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from right with fade none 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 1,
    "transform": "translate3d(100%, 0, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from right with fade out 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 0,
    "transform": "translate3d(100%, 0, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from top with fade in 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 1,
    "transform": "translate3d(0, -100%, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from top with fade inout 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 0,
    "transform": "translate3d(0, -100%, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from top with fade none 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 1,
    "transform": "translate3d(0, -100%, 0)",
  },
}
`;

exports[`<SlideIn /> should animate exiting from top with fade out 1`] = `
{
  "0%": {
    "opacity": 1,
    "transform": "none",
  },
  "100%": {
    "opacity": 0,
    "transform": "translate3d(0, -100%, 0)",
  },
}
`;
